service/src/main/kotlin/io/provenance/api/frameworks/web/misc/ServerRequestExt.kt

```diff
@@ -17,12 +20,23 @@ suspend inline fun <reified T> ServerRequest.requireBody(): T =
         requireNotNull(awaitBodyOrNull()) { "Failed to parse request body of type ${T::class}" }
     }
 
-fun ServerRequest.getUser(): UUID {
+fun ServerRequest.getUser(): UserID {
     try {
-        val header = requireNotNull(headers().firstHeader("x-uuid"))
-        return UUID.fromString(header)
+        return requireNotNull(getUUIDOrNull() ?: getAddressOrNull())
     } catch (exception: IllegalArgumentException) {
-        log.error(exception) { "error parsing x-uuid header" }
-        throw IllegalArgumentException("error parsing x-uuid header", exception)
+        log.error(exception) { "error parsing x-uuid or address header" }
+        throw IllegalArgumentException("error parsing x-uuid or address header", exception)
     }
 }
+
+fun ServerRequest.getUUIDOrNull(): UserUUID? = headers().firstHeader("x-uuid")
+    ?.let { UserUUID(UUID.fromString(it)) }
+
+fun ServerRequest.getAddressOrNull(): UserAddress? {
+    val address = headers().firstHeader("x-figure-tech-address")?.also {
+        log.info { "x-figure-tech-address header = $it" }
+    } ?: headers().firstHeader("x-figure-tech-granter-address")?.also {
+        log.info { "x-figure-tech-granter-address header = $it" }
```

Comment:
looking at both now, will need to get clarification on which is correct. https://hackmd.io/UhGCVFliRYGRSnbqrnDuNg is ambiguous

Reply:
Update?

Reply:
updated to Kong consumer headers